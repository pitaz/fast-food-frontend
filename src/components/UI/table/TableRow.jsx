/* eslint-disable react/jsx-no-bind */
import React, {Fragment} from 'react';
import PropTypes from 'prop-types';
import TableCell from './TableCell.jsx';

const TableRow = (props) => {
  const {orders, action} = props;
  return (
    <Fragment>
    { orders && orders.map(orders => (
        ( 
      <div key={orders.id} className="table-row" id="row">
      <TableCell key={orders.meal} className="table-cell">{orders.meal}</TableCell>
      <TableCell key={orders.price} className="table-cell">{orders.price}</TableCell>
      <TableCell key={orders.quantity} className="table-cell">{orders.quantity}</TableCell>
      <TableCell key={orders.status} className="table-cell">{orders.status}</TableCell>
      <TableCell key={orders.id} className="table-cell">
      <button className='btn-action danger' onClick={() => action.cancelOrder(orders.id)}
      >delete</button>
      </TableCell>
      </div>
      )
      )
    )}
    </Fragment>
  );
};

TableRow.propTypes = {
  orders: PropTypes.instanceOf(Array),
  userId: PropTypes.number,
  action: PropTypes.instanceOf(Object)
};

export default TableRow;
