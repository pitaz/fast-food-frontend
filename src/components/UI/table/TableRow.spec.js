/* eslint-disable react/no-children-prop */
/* eslint-disable no-undef */
import React from 'react';
import expect from 'expect';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import TableRow from './TableRow.jsx';
import configureMockStore from 'redux-mock-store';


const mockStore = configureMockStore();
let wrapper;

describe('The Signin Component Test Suite', () => {
  beforeEach(() => {
    const props = {
      action: {
        fetchMeals: jest.fn(),
      },
      menu: {},
    };
    wrapper = shallow(<TableRow {...props} />);
  });

  it('renders the TableRow component', () => {
    expect(wrapper.exists()).toBe(true);
    expect(wrapper).toMatchSnapshot();
  }); 

  it('should match the snapshot', () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should render <TableRowForm /> Component', () => {
    // expect(wrapper.length).toBe(1);
    expect(wrapper.find(TableRow)).toBeDefined();
  });
});
