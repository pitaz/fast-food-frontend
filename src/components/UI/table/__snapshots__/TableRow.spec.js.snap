// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The Signin Component Test Suite renders the TableRow component 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <TableRow
    action={
      Object {
        "fetchMeals": [MockFunction],
      }
    }
    menu={Object {}}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": undefined,
    },
    "ref": null,
    "rendered": null,
    "type": Symbol(react.fragment),
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": undefined,
      },
      "ref": null,
      "rendered": null,
      "type": Symbol(react.fragment),
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`The Signin Component Test Suite should match the snapshot 1`] = `<Fragment />`;
